src/MusicStore/Program.cs

```diff
@@ -17,14 +17,9 @@ public Task<int> Main(string[] args)
             var config = new Configuration();
             config.AddCommandLine(args);
 
-            var context = new HostingContext()
-            {
-                Configuration = config,
-                ServerFactoryLocation = "Microsoft.AspNet.Server.WebListener",
-                ApplicationName = "MusicStore"
-            };
-
-            using (new HostingEngine().Start(context))
+            using (WebApplication.CreateHostingEngine(config)
```

Comment:
I'm not sure what change you are advocating for music store though,  you don't like how hosting automatically scans for startup today?  You would prefer something more explicit like WebApplication.CreateHostingEngine(config).Start<Startup>()?